app/scripts/lib/transaction/metrics.ts

```diff
@@ -1,44 +1,50 @@
 import { isHexString } from 'ethereumjs-util';
 import EthQuery from 'eth-query';
 import { BigNumber } from 'bignumber.js';
-import type { Provider } from '@metamask/network-controller';
 import { FetchGasFeeEstimateOptions } from '@metamask/gas-fee-controller';
+import { type Provider } from '@metamask/network-controller';
 
-import { ORIGIN_METAMASK } from '../../../shared/constants/app';
+import { ORIGIN_METAMASK } from '../../../../shared/constants/app';
 import {
   determineTransactionAssetType,
   isEIP1559Transaction,
-} from '../../../shared/modules/transaction.utils';
-import { hexWEIToDecGWEI } from '../../../shared/modules/conversion.utils';
+} from '../../../../shared/modules/transaction.utils';
+import {
+  hexWEIToDecETH,
+  hexWEIToDecGWEI,
+} from '../../../../shared/modules/conversion.utils';
 import {
   TransactionType,
   TokenStandard,
   TransactionApprovalAmountType,
   TransactionMetaMetricsEvent,
   TransactionMeta,
-} from '../../../shared/constants/transaction';
+} from '../../../../shared/constants/transaction';
 import {
   MetaMetricsEventCategory,
+  MetaMetricsEventName,
   MetaMetricsEventFragment,
   MetaMetricsPageObject,
   MetaMetricsReferrerObject,
-} from '../../../shared/constants/metametrics';
-import { GasRecommendations } from '../../../shared/constants/gas';
-import { TRANSACTION_ENVELOPE_TYPE_NAMES } from '../../../shared/lib/transactions-controller-utils';
+} from '../../../../shared/constants/metametrics';
+import { GasRecommendations } from '../../../../shared/constants/gas';
+import {
+  calcGasTotal,
+  getSwapsTokensReceivedFromTxMeta,
+  TRANSACTION_ENVELOPE_TYPE_NAMES,
+} from '../../../../shared/lib/transactions-controller-utils';
 ///: BEGIN:ONLY_INCLUDE_IN(blockaid)
 import {
   BlockaidReason,
   BlockaidResultType,
-} from '../../../shared/constants/security-provider';
+} from '../../../../shared/constants/security-provider';
 ///: END:ONLY_INCLUDE_IN
 import {
-  getSnapAndHardwareInfoForMetrics,
   type SnapAndHardwareMessenger,
-} from './snap-keyring/metrics';
-
-export const METRICS_STATUS_FAILED = 'failed on-chain';
+  getSnapAndHardwareInfoForMetrics,
+} from '../snap-keyring/metrics';
 
-export type TransactionMetricsRequest = {
+export type TransactionEventRequest = {
```

Comment:
Is this type not specific to what the metrics handlers need so is `TransactionMetricsRequest` more explicit?

Reply:
Renaming is done [ea4d8df](https://github.com/MetaMask/metamask-extension/pull/21456/commits/ea4d8df9246e32b1dc6c9f79bbada3f2a409a781)